cardano-api/src/Cardano/Api/Fees.hs

```diff
@@ -1098,8 +1098,12 @@ makeTransactionBodyAutoBalance eraInMode systemstart history pparams
                 returnCollateral = fromShelleyLovelace . Ledger.rationalToCoinViaFloor $ amt % 100
 
             case (txReturnCollateral, txTotalCollateral) of
+#if MIN_VERSION_base(4,16,0)
+#else
+              -- TODO figure out how to include the following cases in ghc-9.2.4
               (rc@TxReturnCollateral{}, tc@TxTotalCollateral{}) ->
                 (rc, tc)
+#endif
```

Comment:
We should log the error message here.
```
src/Cardano/Api/Fees.hs:1104:15: error: [-Woverlapping-patterns, -Werror=overlapping-patterns]
    Pattern match is redundant
    In a case alternative:
        (rc@TxReturnCollateral {}, tc@TxTotalCollateral {}) -> ...
     |
1104 |               (rc@TxReturnCollateral{}, tc@TxTotalCollateral{}) ->
     |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...
```

And yes, that one _is_ redundant. It's the same as

https://github.com/input-output-hk/cardano-node/pull/4472/files#diff-12e0bb9f3db22ac71c93ed3932f54c91a6f6bf7243afaa29ccd7ecb3657e6069L1074
```
   calcReturnAndTotalCollateral _ _ _ rc@TxReturnCollateral{} tc@TxTotalCollateral{} _ _ = (rc,tc)
```
which will fire before we even end in this case.